test/lib/jdk/test/lib/crac/CracProcess.java

```diff
@@ -192,4 +194,71 @@ public void sendNewline() throws IOException {
     public void destroyForcibly() {
         process.destroyForcibly();
     }
+
+    public static void printThreadDump(long pid) throws IOException {
+        boolean isAlive = ProcessHandle.of(pid).map(ProcessHandle::isAlive).orElse(false);
+        if (!isAlive) {
+            System.err.println("Process " + pid + " is not alive.");
+        } else {
+            System.err.println("Running: jcmd " + pid + " Thread.print");
+            Process jcmdProc = new ProcessBuilder(jdk.test.lib.Utils.TEST_JDK + "/bin/jcmd", String.valueOf(pid), "Thread.print")
+                    .redirectErrorStream(true)
+                    .start();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(jcmdProc.getInputStream()))) {
+                for (String line = reader.readLine(); null != line; line = reader.readLine()) {
+                    System.err.println("JCMD: " + line);
+                }
+            }
+        }
+    }
+
+    private static boolean checkGcoreAvailable() {
+        ProcessBuilder builder = new ProcessBuilder("which", "gcore");
+        builder.redirectErrorStream(true);
+        try {
+            Process process = builder.start();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line = reader.readLine();
+            int exitCode = process.waitFor();
+            if (exitCode == 0 && line != null && !line.trim().isEmpty()) {
+                System.out.println("gcore is available.");
+                return true;
+            } else {
+                System.out.println("gcore is NOT available.");
+                return false;
+            }
+        } catch (IOException | InterruptedException e) {
+            System.out.println("Could not run 'which gcore' or was interrupted");
+            return false;
+        }
+    }
+
+    public static void dumpProcess(long pid) throws IOException, InterruptedException {
```

Comment:
I believe it does not actually throw these
```suggestion
    public static void dumpProcess(long pid) {
```